cloud.foundry.cli/src/test/java/cloud/foundry/cli/logic/ApplyLogicTest.java

```diff
@@ -391,4 +385,110 @@ public void testApplyServices() {
         verify(entry).getKey();
         verify(entry).getValue();
     }
+
+    @Test
+    public void testApplySpaceWithSpaceNotExistingWillCreateSpace() {
+
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        List<String> presentSpaces = Arrays.asList("space1", "space2");
+        when(spaceOperationsMock.getAll()).thenReturn(Mono.just(presentSpaces));
+
+        Mono<Void> resultingMono = mock(Mono.class);
+        when(spaceOperationsMock.create(desiredSpaceName)).thenReturn(resultingMono);
+
+        // when
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
+        applyLogic.applySpace(desiredSpaceName, spaceOperationsMock);
+
+        // then
+        verify(resultingMono).block();
+    }
+
+    @Test
+    public void testApplySpaceWithSpaceExistingWillNotCreateSpace() {
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        List<String> presentSpaces = Arrays.asList("testName", "space2");
+        when(spaceOperationsMock.getAll()).thenReturn(Mono.just(presentSpaces));
+
+        Mono<Void> resultingMono = mock(Mono.class);
+        when(spaceOperationsMock.create(desiredSpaceName)).thenReturn(resultingMono);
+
+        // when
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
+        applyLogic.applySpace(desiredSpaceName, spaceOperationsMock);
+
+        // then
+        verify(resultingMono, never()).block();
+    }
+
+    @Test
+    public void testApplySpaceWithGetSpaceNamesFailingThrowsGetException() {
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        Mono<List<String>> getRequestMock = mock(Mono.class);
+        when(spaceOperationsMock.getAll()).thenReturn(getRequestMock);
+        when(getRequestMock.block()).thenThrow(new RuntimeException("GetFailing"));
+
+        // when + then
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
```

Comment:
you should move //when + then to this line

Reply:
we have it one line above on porpusoe to emphazise the comment